deploy ERR_PNPM_LOCKFILE_CONFIG_MISMATCH with overrides

### Verify latest release

- [x] I verified that the issue exists in the latest pnpm release

### pnpm version

v10.6.3

### Which area(s) of pnpm are affected? (leave empty if unsure)

Lockfile

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/hartbit/pnpm-deploy-lockfile-overrides

### Reproduction steps

Clone the repository and execute the following commands:

```
$ pnpm i
$ pnpm --filter=a deploy --prod dist
```

### Describe the Bug

PNPM outputs the following error:

```
 ERR_PNPM_LOCKFILE_CONFIG_MISMATCH  Cannot proceed with the frozen installation. The current "overrides" configuration doesn't match the value found in the lockfile
```

### Expected Behavior

I expect the `deploy` not to fail as the `overrides` configuration is identical between `package.json` and `pnpm-lock.yaml` and even if it were, `pnpm i` should guarantee it is the case.

### Which Node.js version are you using?

22.14.0

### Which operating systems have you used?

- [x] macOS
- [ ] Windows
- [ ] Linux

### If your OS is a Linux based, which one it is? (Include the version if relevant)

_No response_

--- a/.changeset/old-symbols-invite.md
+++ b/.changeset/old-symbols-invite.md
@@ -1,5 +1,6 @@
 ---
 "@pnpm/plugin-commands-deploy": patch
+"pnpm": patch
 ---
 
 Let `pnpm deploy` work in repos with `overrides` when `inject-workspace-packages=true` [#9283](https://github.com/pnpm/pnpm/issues/9283).

--- a/.changeset/old-symbols-invite.md
+++ b/.changeset/old-symbols-invite.md
@@ -1,0 +1,5 @@
+---
+"@pnpm/plugin-commands-deploy": patch
+---
+
+Let `pnpm deploy` work in repos with `overrides` when `inject-workspace-packages=true` [#9283](https://github.com/pnpm/pnpm/issues/9283).

--- a/releasing/plugin-commands-deploy/src/createDeployFiles.ts
+++ b/releasing/plugin-commands-deploy/src/createDeployFiles.ts
@@ -119,7 +119,6 @@ export function createDeployFiles ({
   const result: DeployFiles = {
     lockfile: {
       ...lockfile,
-      overrides: undefined, // the effects of package overrides should already be part of the package snapshots
       patchedDependencies: undefined,
       packageExtensionsChecksum: undefined, // the effects of the package extensions should already be part of the package snapshots
       pnpmfileChecksum: undefined, // the effects of the pnpmfile should already be part of the package snapshots
@@ -136,7 +135,6 @@ export function createDeployFiles ({
       pnpm: {
         ...rootProjectManifest?.pnpm,
         ...pick(USEFUL_NON_ROOT_PNPM_FIELDS, selectedProjectManifest.pnpm ?? {}),
-        overrides: undefined, // the effects of package overrides should already be part of the package snapshots
         patchedDependencies: undefined,
         packageExtensions: undefined, // the effects of the package extensions should already be part of the package snapshots
       },

--- a/releasing/plugin-commands-deploy/test/shared-lockfile.test.ts
+++ b/releasing/plugin-commands-deploy/test/shared-lockfile.test.ts
@@ -337,9 +337,9 @@ test('the deploy manifest should inherit the pnpm object from the root manifest 
   const manifest = readPackageJson('deploy') as ProjectManifest
   expect(manifest.pnpm).toStrictEqual({
     onlyBuiltDependencies: preparedManifests.root.pnpm!.onlyBuiltDependencies,
+    overrides: preparedManifests.root.pnpm!.overrides,
     executionEnv: preparedManifests['project-0'].pnpm!.executionEnv,
   } as ProjectManifest['pnpm'])
-  expect(manifest.pnpm?.overrides).toBeUndefined() // by design
 
   expect(readPackageJson('deploy/node_modules/is-positive/')).toHaveProperty(['version'], preparedManifests.root.pnpm!.overrides!['is-positive'])
   expect(project.readLockfile().importers).toStrictEqual({

